src/common/graph.py

```diff
@@ -0,0 +1,43 @@
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+from graphql import DocumentNode
+
+from src.common.decorators import retry_gql_errors
+from src.config.settings import DEFAULT_RETRY_TIME, GRAPH_API_TIMEOUT, settings
+
+GRAPH_PAGE_SIZE = 100
+
+
+class GraphClient:
+    gql_client: Client
+
+    def __init__(self):
+        transport = AIOHTTPTransport(
+            url=settings.network_config.STAKEWISE_API_URL,
+            timeout=GRAPH_API_TIMEOUT,
+        )
+        self.gql_client = Client(transport=transport)
+
+    @retry_gql_errors(delay=DEFAULT_RETRY_TIME)
+    async def run_query(self, query: DocumentNode, params: dict | None = None) -> dict:
+        result = await self.gql_client.execute_async(query, variable_values=params)
+        return result
+
+    async def get_vault_validators(self, vault: str) -> list[str]:
+        query = gql(
+            """
+            query Validators($vaultAddress: String!) {
+              vaultValidators(
```

Comment:
There is a limit on number of validators in a single query. See `first` param. It is 100 by default. Need to use pagination.